src/twisted/protocols/_smb/core.py

```diff
@@ -0,0 +1,760 @@
+# -*- test-case-name: twisted.protocols._smb.tests -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+"""Implement Microsoft's Server Message Block protocol"""
+
+import binascii
+import enum
+import struct
+import time
+from collections import namedtuple
+from typing import Any, Callable, Optional, Sequence, Tuple, Union
+from uuid import uuid4
+
+import attr
+
+from twisted.cred.checkers import ANONYMOUS
+from twisted.cred.portal import Portal
+from twisted.internet import protocol
+from twisted.internet.defer import maybeDeferred
+from twisted.internet.interfaces import IAddress
+from twisted.logger import Logger
+from twisted.protocols._smb import _base, security_blob
+from twisted.protocols._smb._base import byte, long, medium, octets, short, uuid
+from twisted.protocols._smb.ismb import (
+    IFilesystem,
+    IPipe,
+    IPrinter,
+    ISMBServer,
+    NoSuchShare,
+)
+from twisted.python.failure import Failure
+
+log = Logger()
+
+SMBMind = namedtuple("SMBMind", "session_id domain addr")
+
+
+@attr.s
+class NegReq:
+    """negotiate request"""
+
+    size = short(36, locked=True)
+    dialect_count = short()
+    security_mode = short()
+    reserved = short()
+    capabilities = medium()
+    client_uuid = uuid()
+
+
+MAX_READ_SIZE = 0x10000
+MAX_TRANSACT_SIZE = 0x10000
+MAX_WRITE_SIZE = 0x10000
+
+
+@attr.s
+class NegResp:
+    """negotiate response"""
+
+    size = short(65, locked=True)
+    signing = short()
+    dialect = short()
+    reserved = short()
+    server_uuid = uuid()
+    capabilities = medium()
+    max_transact = medium(MAX_TRANSACT_SIZE)
+    max_read = medium(MAX_READ_SIZE)
+    max_write = medium(MAX_WRITE_SIZE)
+    time = long()
+    boot_time = long()
+    offset = short(128, locked=True)
+    buflen = short()
+    reserved2 = medium()
```

Comment:
Given that the respective functions here all return `Any`, they should have type annotations so that they get type checked.